### PR TITLE
Bit of refactoring + offline mode for the sdk

### PR DESCRIPTION
**Pull request by oserban**

### CODE REVIEW

**Comment by oserban** [comment]
Spelling error in the commit message. It was a bit of refactoring ...

File: sdks/python/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+certifi==2018.8.24
```
Comment (jamesscottbrown) [inline sdks/python/requirements.txt]: Why are all of these packages listed here? Many of them seem unnecessary.

Comment (jamesscottbrown) [inline sdks/python/requirements.txt]: Why are all of these packages listed here? Many of them seem unnecessary.

**Comment by oserban** [comment]
I made the changes requested

**Pull request merged**